### PR TITLE
WIP: Ovirt_disk - Add ability to refresh passthrough disks

### PR DESCRIPTION
**Pull request by rhodain**
##### SUMMARY

A new parameter "host" is added. When this parameter is specified the newly created disk or an existing disk will refresh its information about the underlying storage via the specified host. This option is only valid for passthrough disks. This option requires at least the logical_unit.id to be specified.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ovirt_disk
##### ANSIBLE VERSION

```paste below
ansible 2.6.3
```

##### ADDITIONAL INFORMATION

./hacking/test-module -m ./lib/ansible/modules/cloud/ovirt/ovirt_disk.py -a "name=MyDisk01 logical_unit=id=36001405fb6247061a564688a9baa34dba  host=RHV-H01"

```paste below

```

### CODE REVIEW

**Comment by mscherer** [comment]
This PR would conflict with https://github.com/ansible/ansible/pull/45643 that you opened before :/

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py [hunk 1]
```diff
@@ -578,12 +586,23 @@ def main():
...
         supports_check_mode=True,
     )
 
+    lun = module.params.get('logical_unit')
+    host = module.params['host']
+    # Fail when host is specified with the LUN id. Lun id is needed to identify
+    # an existing disk if already available inthe environment.
+    if host and lun.get("id") is None:
+        module.fail_json(
+            msg="Can not use parameter host ({0!s}) withouti "
```
Comment (mscherer) [inline lib/ansible/modules/cloud/ovirt/ovirt_disk.py]: I guess the "i" should be removed, or exchanged for a "s" to go with "pecifying" after

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py [hunk 2]
```diff
@@ -693,6 +713,19 @@ def main():
             elif state == 'detached':
                 ret = disk_attachments_module.remove()
 
+        # When host is specified and the disk is not being removed refresh the
+        # information abou thte LUN.
```
Comment (mscherer) [inline lib/ansible/modules/cloud/ovirt/ovirt_disk.py]: Nit picking but while on it, can you correct the comment ?

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py [hunk 3]
```diff
@@ -693,6 +713,19 @@ def main():
...
+        host = module.params.get('host')
```
Comment (mscherer) [inline lib/ansible/modules/cloud/ovirt/ovirt_disk.py]: Duplicate from line 597

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py [hunk 4]
```diff
@@ -146,6 +146,14 @@
                your playbook accordingly to not export the disk all the time.
                This option is valid only for template disks."
         version_added: "2.4"
+    host:
+        description:
+            - "When the hypervisor name is specified the newly created disk or
+               an existing disk will refresh its information about the
+               underlying storage. This oprion is only valid for passthrough
```
Comment (mscherer) [inline lib/ansible/modules/cloud/ovirt/ovirt_disk.py]: s/oprion/option/

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py [hunk 5]
```diff
@@ -146,6 +146,14 @@
                your playbook accordingly to not export the disk all the time.
                This option is valid only for template disks."
         version_added: "2.4"
+    host:
```
Comment (mscherer) [inline lib/ansible/modules/cloud/ovirt/ovirt_disk.py:149]: Could a more descriptive name be given ? I am not sure people would assume that giving the host will trigger a refresh (at least I don't, but I do not know ovirt well enough).

**Review by mscherer — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py (hunk 1 above)
Comment (rhodain) [reply lib/ansible/modules/cloud/ovirt/ovirt_disk.py]: Corrected

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py (hunk 2 above)
Comment (rhodain) [reply lib/ansible/modules/cloud/ovirt/ovirt_disk.py]: Corrected

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py (hunk 3 above)
Comment (rhodain) [reply lib/ansible/modules/cloud/ovirt/ovirt_disk.py]: You are right, so I have removed this line.

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py (hunk 4 above)
Comment (rhodain) [reply lib/ansible/modules/cloud/ovirt/ovirt_disk.py]: Corrected

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py (hunk 5 above)
Comment (rhodain) [reply lib/ansible/modules/cloud/ovirt/ovirt_disk.py:149]: I added some additional information.

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py [hunk 6]
```diff
@@ -146,6 +146,16 @@
...
+        description:
+            - "When the hypervisor name is specified the newly created disk or
+               an existing disk will refresh its information about the
+               underlying storage( Disk size, Serial, Product ID, Vendor ID ...)
+               The specified host will be used for garherting the storage
```
Comment (mscherer) [inline lib/ansible/modules/cloud/ovirt/ovirt_disk.py]: I think 'garherting' is not the spelling you want

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py (hunk 5 above)
Comment (gundalow) [reply lib/ansible/modules/cloud/ovirt/ovirt_disk.py:149]: Is `host` the term that ovirt uses?
Just want to ensure their isn't confusion between what Ansible refers to as a host vs ovirt

Comment (mscherer) [reply lib/ansible/modules/cloud/ovirt/ovirt_disk.py:149]: Yeah, but so, could it be renamed, or does it make sense for ovirt users ? Again, i am sysadmin and I find that naming to be confusing (as I guess "host" is used for the host to connect usually) , but I am not using ovirt, so maybe it is obvious for people who are using it.

**Comment by mscherer** [comment]
Also, the module say it support check_mode, but it does seems to be the case at all :/

**Comment by machacekondra** [comment]
> Also, the module say it support check_mode, but it does seems to be the case at all :/

It does. The logic is mostly in the `module_utils/ovirt.py`. But there may be cases where it's not supported, but it's an issue and we need to fix it.

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py (hunk 5 above)
Comment (mwperina) [reply lib/ansible/modules/cloud/ovirt/ovirt_disk.py:149]: It's correct for oVirt/RHV, by host we mean a machine where VMs are running.

File: lib/ansible/modules/cloud/ovirt/ovirt_disk.py (hunk 6 above)
Comment (mwperina) [reply lib/ansible/modules/cloud/ovirt/ovirt_disk.py]: Roman, you probably wanted to use 'gathering', right?

**Comment by gundalow** [comment]
Hi if you have a look at http://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html that should help you with rebasing this PR

**Pull request closed without merging**